### DISCUSSION TITLE
How to run a loop for a set of Videos and Audios in a trial?

### DISCUSSION THREAD

**Discussion by Mananshi** [jspsych/jsPsych#1844] [Q&A]
I am currently trying to design an experiment wherein I have to present a video/audio which would be followed by a question and record participant's response via the screen or the keyboard. The participant is not required to interact with any of the media files and has to just listen to the audio or view the video and answer a simple question for which, choices would be pre-defined.

Note: Numerous Videos and Audio are to be presented randomly hence creating individual stimuli won't be feasible.

Can someone please help me out with this? I am a beginner and I am not quite aware of how to work with Video and Audio files in a trial.

**Comment by jodeleeuw** [comment]
Hi @Mananshi,

If you want to present a video file without interaction, then you can use the `video-keyboard-response` plugin and set the `choices` parameter to `jsPsych.NO_KEYS`. This will prevent the user from interacting with the trial. You can combine this with either `trial_ends_after_video` or `trial_duration` to control the timing of the end of the trial.

For audio files the process is the same, but with the `audio-keyboard-response` plugin.

**Reply by Mananshi** [reply]
Noted. Thank you very much for your help!
